src/core_landice/mode_forward/mpas_li_iceshelf_melt.F

```diff
@@ -1253,63 +1266,65 @@ subroutine grounded_face_melt_ismip6(domain, err)
-                    .or. ( li_mask_is_floating_ice(cellMask(iNeighbor)) & 
-                    .and. li_mask_is_margin(cellMask(iNeighbor)) &
-                    .and. (.not. li_mask_is_albany_active(cellMask(iNeighbor))) ) ) then
-                     nEmptyNeighbors = nEmptyNeighbors + 1
-                     oceanEdgeLength = oceanEdgeLength + dvEdge(edgesOnCell(iEdge, iCell))
+            
+            ! Calculate length of grounding line for each cell
+            if ( bedTopography(iCell) < seaLevel &
+                .and. li_mask_is_grounded_ice(cellMask(iCell)) ) then
+               do iEdge = 1, nEdgesOnCell(iCell)
+                 iNeighbor = cellsOnCell(iEdge, iCell)
+                 if ( li_mask_is_grounding_line(edgeMask(edgesOnCell(iEdge,iCell))) ) then 
+                    oceanEdgeLength = oceanEdgeLength + dvEdge(edgesOnCell(iEdge, iCell))
                  endif
-            enddo 
+               enddo 
+            endif
             
-            if ( nEmptyNeighbors > 0 &
-               .and. li_mask_is_grounded_ice(cellMask(iCell)) &
-               .and. bedTopography(iCell) < seaLevel ) then
-                
-                groundedMarineMarginMask(iCell) = 1
+            if ( bedTopography(iCell) < 0.0_RKIND ) then
                 waterDepth = seaLevel - bedTopography(iCell)
-                          
-                submergedFaceArea = waterDepth * oceanEdgeLength
+            else
+                waterDepth = 0.0
+            endif
+           
+            submergedFaceArea = waterDepth * oceanEdgeLength
                 
-                if (trim(config_front_mass_bal_grounded) == 'ismip6') then                
+            if (trim(config_front_mass_bal_grounded) == 'ismip6') then                
                 ! Calculate ice front melt rate at each cell
-                   faceMeltSpeed(iCell) = (aSubglacial * waterDepth * & ! m s^-1
+               faceMeltSpeed(iCell) = (aSubglacial * waterDepth * & ! m s^-1
                          (ismip6Runoff(iCell) / rhow * secPerDay)**alphaSubglacial &
-                         + B) * (TFocean(iCell) + addTFocean)**betaTF / secPerDay
-                elseif (trim(config_front_mass_bal_grounded) == 'uniform') then   
+                         + B) *  max(0.0_RKIND, TFocean(iCell) + addTFocean)**betaTF / secPerDay
+            elseif (trim(config_front_mass_bal_grounded) == 'uniform') then   
                    faceMeltSpeed(iCell) = config_uniform_face_melt_rate
-                endif
-                faceMeltRateApplied(iCell) = faceMeltSpeed(iCell) * submergedFaceArea ! m^3 s^-1
-            else 
-                groundedMarineMarginMask(iCell) = 0
-                faceMeltRateApplied(iCell) = 0.0_RKIND     
-                faceMeltSpeed(iCell) = 0.0_RKIND
-            endif           
+            endif
+
          enddo
 
+         call mpas_log_write("calling apply_front_ablation_velocity")
```

Comment:
Add something like "...from grounded_face_melt_ismip6" to this log message.

Reply:
Added